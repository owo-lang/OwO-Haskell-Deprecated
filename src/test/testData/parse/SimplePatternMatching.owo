module SimplePatternMatching where

  onlyVariables : SomeFunctionType
  onlyVariables a b c = f a b c

  destructed : SomeFunctionType
  destructed (f a) (g b) = destructed (g a) (f b)

  noParenthesis : SomeFunctionType
  noParenthesis f a b = noParenthesis (f b a)
