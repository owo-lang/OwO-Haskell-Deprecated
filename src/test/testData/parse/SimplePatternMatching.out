File type: CodeFileType
Top module name: .SimplePatternMatching
 type signature "onlyVariables" (38 3 3) (51 3 16)
  no pragmas
  named reference "SomeFunctionType" (54 3 19) (70 3 35)
 implementation of function  "onlyVariables" (73 4 3) (86 4 16)
  no pragmas
  clause for  "onlyVariables" (73 4 3) (86 4 16)
  patterns
   function application
   function being applied
    function application
    function being applied
     named reference "a" (87 4 17) (88 4 18)
    value applied to the function
     named reference "b" (89 4 19) (90 4 20)
   value applied to the function
    named reference "c" (91 4 21) (92 4 22)
  function body
   function application
   function being applied
    function application
    function being applied
     function application
     function being applied
      named reference "f" (95 4 25) (96 4 26)
     value applied to the function
      named reference "a" (97 4 27) (98 4 28)
    value applied to the function
     named reference "b" (99 4 29) (100 4 30)
   value applied to the function
    named reference "c" (101 4 31) (102 4 32)
  no with abstraction
  no where clause
 type signature "destructed" (106 6 3) (116 6 13)
  no pragmas
  named reference "SomeFunctionType" (119 6 16) (135 6 32)
 implementation of function  "destructed" (138 7 3) (148 7 13)
  no pragmas
  clause for  "destructed" (138 7 3) (148 7 13)
  patterns
   function application
   function being applied
    function application
    function being applied
     named reference "f" (150 7 15) (151 7 16)
    value applied to the function
     named reference "a" (152 7 17) (153 7 18)
   value applied to the function
    function application
    function being applied
     named reference "g" (156 7 21) (157 7 22)
    value applied to the function
     named reference "b" (158 7 23) (159 7 24)
  function body
   function application
   function being applied
    function application
    function being applied
     named reference "destructed" (163 7 28) (173 7 38)
    value applied to the function
     function application
     function being applied
      named reference "g" (175 7 40) (176 7 41)
     value applied to the function
      named reference "a" (177 7 42) (178 7 43)
   value applied to the function
    function application
    function being applied
     named reference "f" (181 7 46) (182 7 47)
    value applied to the function
     named reference "b" (183 7 48) (184 7 49)
  no with abstraction
  no where clause
 type signature "noParenthesis" (189 9 3) (202 9 16)
  no pragmas
  named reference "SomeFunctionType" (205 9 19) (221 9 35)
 implementation of function  "noParenthesis" (224 10 3) (237 10 16)
  no pragmas
  clause for  "noParenthesis" (224 10 3) (237 10 16)
  patterns
   function application
   function being applied
    function application
    function being applied
     named reference "f" (238 10 17) (239 10 18)
    value applied to the function
     named reference "a" (240 10 19) (241 10 20)
   value applied to the function
    named reference "b" (242 10 21) (243 10 22)
  function body
   function application
   function being applied
    named reference "noParenthesis" (246 10 25) (259 10 38)
   value applied to the function
    function application
    function being applied
     function application
     function being applied
      named reference "f" (261 10 40) (262 10 41)
     value applied to the function
      named reference "b" (263 10 42) (264 10 43)
    value applied to the function
     named reference "a" (265 10 44) (266 10 45)
  no with abstraction
  no where clause
